diff --git a/src/main/java/com/aparapi/examples/life/Main.java b/src/main/java/com/aparapi/examples/life/Main.java
index c95c404..20c6a38 100644
--- a/src/main/java/com/aparapi/examples/life/Main.java
+++ b/src/main/java/com/aparapi/examples/life/Main.java
@@ -214,9 +214,9 @@ public class Main{
    public static void main(String[] _args) {
 
       final JFrame frame = new JFrame("Game of Life");
-      final int width = Integer.getInteger("width", 1024 + 512 + 256 + 128);
+      final int width = Integer.getInteger("width", 2048+1024);
 
-      final int height = Integer.getInteger("height", 768 + 256);
+      final int height = Integer.getInteger("height", 1536+256);
 
       // Buffer is twice the size as the screen.  We will alternate between mutating data from top to bottom
       // and bottom to top in alternate generation passses. The LifeKernel will track which pass is which
diff --git a/src/main/java/com/aparapi/examples/mandel/Main.java b/src/main/java/com/aparapi/examples/mandel/Main.java
index fa3c57d..b063002 100644
--- a/src/main/java/com/aparapi/examples/mandel/Main.java
+++ b/src/main/java/com/aparapi/examples/mandel/Main.java
@@ -85,7 +85,7 @@ import java.util.List;
 public class Main{
 
    static {
-      System.setProperty("com.aparapi.dumpProfilesOnExit", "true");
+ //     System.setProperty("com.aparapi.dumpProfilesOnExit", "true");
 //      KernelManager.setKernelManager(new KernelManager() {
 //         @Override
 //         protected List<Device.TYPE> getPreferredDeviceTypes() {
@@ -210,10 +210,10 @@ public class Main{
       final JFrame frame = new JFrame("MandelBrot");
 
       /** Width of Mandelbrot view. */
-      final int width = 768;
+      final int width = 768*2;
 
       /** Height of Mandelbrot view. */
-      final int height = 768;
+      final int height = 768*2;
 
       /** Mandelbrot image height. */
       final Range range = Range.create(width * height);
